contracts/PBAB+Collabs/GenArt721CoreV2WithEvents.sol

```diff
@@ -0,0 +1,793 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../interfaces/0.8.x/IRandomizer.sol";
+import "../interfaces/0.8.x/IGenArt721CoreV2WithEvents.sol";
+
+import "@openzeppelin-4.5/contracts/utils/Strings.sol";
+
+import "@openzeppelin-4.5/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+
+pragma solidity 0.8.9;
+
+/**
+ * @title Powered by Art Blocks ERC-721 core contract.
+ * @author Art Blocks Inc.
+ */
+contract GenArt721CoreV2WithEvents is
+    ERC721Enumerable,
+    IGenArt721CoreV2WithEvents
+{
+    /// randomizer contract
+    IRandomizer public randomizerContract;
+
+    // generic platform event fields
+    bytes32 constant FIELD_ARTBLOCKS_OWNER = "owner";
+    bytes32 constant FIELD_RENDER_PROVIDER_ADDRESS = "renderProviderAddress";
+    bytes32 constant FIELD_RANDOMIZER_ADDRESS = "randomizerAddress";
+    bytes32 constant FIELD_RENDER_PROVIDER_PERCENTAGE =
+        "renderProviderPercentage";
+    // generic project event fields
+    bytes32 constant FIELD_PROJECT_ACTIVE = "active";
+    bytes32 constant FIELD_ARTIST_ADDRESS = "artistAddress";
+    bytes32 constant FIELD_PROJECT_PAUSED = "paused";
+    bytes32 constant FIELD_PROJECT_CREATED = "created";
+    bytes32 constant FIELD_PROJECT_NAME = "name";
+    bytes32 constant FIELD_PROJECT_ARTIST_NAME = "artistName";
+    bytes32 constant FIELD_SECONDARY_MARKET_ROYALTY_PERCENTAGE =
+        "secondaryMarketRoyaltyPercentage";
```

Comment:
This string takes up all 32 bytes.
on-chain, that should be fine, but ethers doesn't like formatting strings as bytes32 when they are >31 bytes (ref: https://docs.ethers.io/v5/api/utils/strings/#utils-formatBytes32).
Thus, what do you think about staying from edge cases here and just make this `"royaltyPercentage"`?